src/main/java/com/rtb/projectmanagementtool/taskblocker/TaskBlockerController.java

```diff
@@ -25,58 +29,88 @@ public void addTaskBlocker(long taskID, long blockerID) {
   }
 
   public void addTaskBlocker(TaskBlockerData taskBlocker) {
-    // Get task
-    TaskData task;
-    try {
-      task = taskController.getTaskByID(taskBlocker.getTaskID());
-    } catch (NullPointerException e) {
-      System.out.println("Cannot find task with provided taskID");
+    // Ensure the tasks exist
+    ArrayList<TaskData> tasks =
+        taskController.getTasksByIDs(
+            new ArrayList<>(Arrays.asList(taskBlocker.getTaskID(), taskBlocker.getBlockerID())));
+    if (tasks.size() < 2) {
+      System.out.println("Cannot find tasks with provided taskID or blockerID");
       return;
     }
-    // Get blocker
-    TaskData blocker;
-    try {
-      blocker = taskController.getTaskByID(taskBlocker.getBlockerID());
-    } catch (NullPointerException e) {
-      System.out.println("Cannot find task with provided blockerID");
+    // Ensure the blocked task isn't already set to COMPLETE
+    if (tasks.get(0).getStatus() == Status.COMPLETE
+        || tasks.get(1).getStatus() == Status.COMPLETE) {
+      System.out.println("One or more tasks are already set to COMPLETE.");
       return;
     }
-    // Add taskBlocker
-    if (task.getStatus() != Status.COMPLETE) {
-      TransactionOptions options = TransactionOptions.Builder.withXG(true);
-      Transaction transaction = datastore.beginTransaction(options);
-      try {
-        datastore.put(taskBlocker.toEntity());
-        assert containsCycle() == false;
-        transaction.commit();
-      } finally {
-        if (transaction.isActive()) {
-          transaction.rollback();
-        }
-      }
-    } else {
-      System.out.println("Cannot block a task set to COMPLETE.");
+    // Ensure a cycle wouldn't be created if the TaskBlocker is added
+    if (containsPath(taskBlocker.getBlockerID(), taskBlocker.getTaskID())) {
+      System.out.println("Cannot block a task if it would create a cycle.");
+      return;
     }
+    // Add the TaskBlocker
+    taskBlocker.setTaskBlockerID(datastore.put(taskBlocker.toEntity()).getId());
   }
 
-  private boolean containsCycle() {
-    // TODO: Implement cycle detection algorithm
+  private boolean containsPath(long start, long end) {
+    HashSet<TaskBlockerData> taskBlockers = getAllTaskBlockers();
+    Map<Long, Boolean> taskBlockersVisited =
+        taskBlockers
+            .stream()
```

Comment:
The Java presubmit test is failing because it wants to combine these two lines onto one line, but they automatically split up again whenever the code is run.

Reply:
The coveo plugin in pom.xml is using version 2.5.0 which was released in 2018, while the java formatter used by the [presubmit checks](https://github.com/googleinterns/step171-2020/blob/master/.github/workflows/javalint.yaml) uses version 1.8, which was released in May 2020.  Maybe a rule was updated recently and the presubmit is running newer rules, whole the pom.xml plugin is running older rules.  Try a newer version of the coveo plugin:  https://github.com/coveooss/fmt-maven-plugin/releases